### PR TITLE
Fix compilation on LLVM 3.9

### LINKED ISSUE #15582: JIT function lookup broken on LLVM svn
The backtrace (pc pointers) is good but the function name/info lookup for JIT functions are broken.

It seems to be broken by https://github.com/JuliaLang/julia/pull/15519 @vtjnash

### PR DESCRIPTION
**Pull request by yuyichao**
`ErrorOr` business again....

Closes #15582

### CODE REVIEW

**Comment by yuyichao** [comment]
Also fix the debug info registration by keeping a temporary map to find the loaded section corresponding to a debug section. @vtjnash

**Comment by vtjnash** [comment]
Lgtm

File: src/debuginfo.cpp [hunk 1]
```diff
@@ -207,6 +207,15 @@ struct revcomp {
     { return lhs>rhs; }
 };
 
+#ifdef LLVM38
+struct strrefcomp {
+    bool operator() (const StringRef& lhs, const StringRef& rhs) const
+    {
+        return lhs.compare(rhs) > 0;
```
Comment (yuyichao) [inline src/debuginfo.cpp]: Ooops, the order was copied from the `revcomp` above and is reversed... Shouldn't matter so I'll just leave it like this...

**Comment by yuyichao** [comment]
AppVeyor failure is the intermittent threads test timing issue. FWIW, this PR doesn't change any code built by the CIs...

**Pull request merged**